corehq/apps/hqwebapp/static/hqwebapp/js/ckeditor_knockout_bindings.js

```diff
@@ -0,0 +1,185 @@
+                            "forwardDelete",
+                            "delete",
+                            "bold",
+                            "italic",
+                            "enter",
+                            "selectAll",
+                            "shiftEnter",
+                            "insertText",
+                            "input",
+                            "undo",
+                            "redo",
+                            "fontFamily",
+                            "fontSize",
+                            "paragraph",
+                            "insertParagraph",
+                            "heading",
+                            "horizontalLine",
+                            "insertImage",
+                            "replaceImageSource",
+                            "imageInsert",
+                            "imageTextAlternative",
+                            "imageTypeInline",
+                            "toggleImageCaption",
+                            "imageStyle",
+                            "resizeImage",
+                            "imageResize",
+                            "uploadImage",
+                            "imageUpload",
+                            "indent",
+                            "outdent",
+                            "link",
+                            "unlink",
+                            "numberedList",
+                            "bulletedList",
+                            "indentList",
+                            "outdentList",
+                        ],
+                    },
+                },
+                editorInstance = undefined;
+
+            CKEditor5.editorClassic.ClassicEditor.create(element, options).then(function (editor) {
+                var isSubscriberChange = false,
+                    isEditorChange = false,
+                    editorInstance = editor;
+                if (typeof ko.utils.unwrapObservable(valueAccessor()) !== "undefined") {
+                    editorInstance.setData(ko.utils.unwrapObservable(valueAccessor()));
+                }
+
+                // Update the observable value when the document changes
+                editorInstance.model.document.on('change:data', function () {
+                    if (!isSubscriberChange) {
+                        isEditorChange = true;
+                        valueAccessor()(editorInstance.getData());
+                        isEditorChange = false;
+                    }
+                    
+                });
+
+                // Update the document whenever the observable changes
+                valueAccessor().subscribe(function (value) {
+                    if (!isEditorChange) {
+                        isSubscriberChange = true;
+                        editorInstance.setData(value);
+                        isSubscriberChange = false;
+                    }
+
+                });
+
+                if (initialPageData.get('read_only_mode')) {
```

Comment:
Would it be possible to set this flag as an HTML data attribute or would that risk manipulation?  If this editor appears on the same page as other fields it might be nice to have this variable's scope a bit more explicit.

Reply:
This was not defined by me, and is used elsewhere in the javascript in the messaging code. 

https://github.com/dimagi/commcare-hq/blob/6fc82f40dba1973d526483db2b3b4ac3670ab7d4/corehq/apps/data_interfaces/static/data_interfaces/js/case_rule_main.js#L7

https://github.com/dimagi/commcare-hq/blob/6fc82f40dba1973d526483db2b3b4ac3670ab7d4/corehq/apps/data_interfaces/static/data_interfaces/js/make_read_only.js#L6-L10

It seems to me like this pattern is used in other places in HQ too, and this happens in places where you wouldn't be able to submit the form anyway. 

I also don't think there is risk of manipulation here, as in any places where the fields aren't editable, the backend should be validating that there aren't fields being submitted.